src/components/ChartJsWrapper/index.jsx

```diff
@@ -33,50 +34,70 @@ const ChartJsWrapper = ({
   type,
   chartHeight,
   spinnerSize,
-  missingDataError = false,
-}) => {
-  let showError;
+}) =>
+  (() => {
+    if (!data) {
+      return (
+        <div
+          style={{
+            lineHeight: spinnerSize,
+            textAlign: 'center',
+            width: spinnerSize,
+          }}>
+          <CircularProgress />
+        </div>
+      );
+    }
+
+    let error = null;
 
-  if (data && missingDataError) {
-    showError = data.datasets.every(dataset => {
-      const latestDataDate = new Date(dataset.data[dataset.data.length - 1].x);
+    data.datasets.forEach(dataset => {
+      const latestDataDate = new Date(
+        frum(dataset.data)
```

Comment:
I know you love your chaining library but it is not adding a lot more value and it forces new users (and old users like me) to have to learn a new way of doing things rather than using standard data manipulations practices.

I believe there's more value to not switch to it for the sake of contributions.

Reply:
For the future, when you suggest "standard data manipulations practices", please include the code you would like to see.  I do not know what you mean by "standard".

Instead of this

      const latestDataDate = new Date(
        frum(dataset.data)
          .sort('x')
          .last().x
      );

use this

      const sortedTimestamps = dataset.data.map(({x}) => x).sort();
      const latestDataDate = new Date(sortedTimestamps[sortedTimestamps.length-1]);

or this

      const latestDataDate = new Date(
        dataset.data
          .map(({x}) => x)
          .sort()
          .reverse()[0]
      );

or this

      const latestDataDate = new Date(Math.max(dataset.data.map(({ x }) => x)));

which is the nicest.